# Mini-projet - Récupérer des méta-données EXIF d'une image

Pour commencer on importe le module nécessaire

In [1]:
# pour travailler sur des images numériques
from PIL import Image

## Récupération des méta-données

In [2]:
# on charge l'image
photo0 = Image.open("chouchou0.jpg")
# on récupère les méta-données
meta0 = photo0._getexif()

1. Ces méta-données sont stockées sous la forme d'un dictionnaire.

In [3]:
# On peut lire ces données avec la boucle POUR sur les clefs et valeurs.
for clef, valeur in meta0.items() :
    print(clef, " : ", valeur)

36864  :  b'0220'
37121  :  b'\x01\x02\x03\x00'
37377  :  8.724502563476562
36867  :  2019:09:19 16:26:26
36868  :  2019:09:19 16:26:26
37378  :  2.274993896484375
37379  :  6.0
37380  :  0.0
37381  :  2.274993896484375
37383  :  2
37384  :  0
37385  :  24
37386  :  3.3000030517578125
40961  :  65535
40962  :  3264
41990  :  0
40963  :  1836
41992  :  0
41993  :  0
41994  :  0
41996  :  0
271  :  SAMSUNG
272  :  SM-G531F
41495  :  2
274  :  1
531  :  1
33434  :  0.002364066193853428
282  :  72.0
283  :  72.0
33437  :  2.2
41729  :  b'\x01'
34850  :  2
34853  :  {1: 'N', 2: (47.0, 27.0, 59.0), 3: 'W', 4: (0.0, 33.0, 5.0), 6: 0.0}
34855  :  50
296  :  2
41986  :  0
40960  :  b'0100'
41987  :  0
305  :  GIMP 2.10.8
306  :  2019:09:29 19:26:26
41988  :  1.0
41989  :  31
34665  :  212
37500  :  b'\x07\x00\x02\x00\x07\x00\x01\x00\x00\x00\x00\x00\x00\x00\x0c\x00\x07\x00\x01\x00\x00\x00\x00\x00\x00\x00@\x00\x07\x00\x01\x00\x00\x00\x00\x00\x00\x00P\x00\x07\x00\x01\x00\x00\x00\x01\x00\x00\x00`\x

2. Le problème c'est que les clefs du dictionnaire sont peu lisibles. On peut néanmoins déviner quelques résultats.<br />
Quelle est la clef correspondant à la marque de l'appareil utilisé pour faire cette photo ? <br />
*Réponse : C'est la clef `271`.*<br />
Quelle est la clef correspondant aux coordonénes géographiques ? <br />
*Réponse : C'est la clef `34853`.*<br />
Que peut-on dire de la valeur associé à la clef `37500` ? <br />
*Réponse : La réponse est donnée en hexadécimale.*<br />

3. La clef `274` donne l'orientation de la photo suivant le schéma ci-dessous : <br/>
![](orientation.png) <br />
Quelle est l'orientation de la photo `chouchou0.jpg` ?

In [4]:
meta0[274]

1

*Réponse : 1, autrement dit, format paysage.*

4. Même chose pour la photo `chouchou1.jpg`.

In [5]:
# on procède comme au départ avec la première image :
# il faut charger l'image
photo1 = Image.open("chouchou1.jpg")
# on récupère les méta-données
meta1 = photo1._getexif()
#on interroge la bonne clef
meta1[274]

6

*Réponse : 6, autrement dit, format portrait.*